Fix a few warnings

neheb opened the pull request:
The second is technically incorrect but not as verbose as __attribute__((unused))

neheb commented:
One benefit is that the dependency on gnu99 can now be removed.

aircrack-ng commented:
It currently breaks compilation

neheb commented:
Yeah the second commit is faulty. I'll remove it soon.

aircrack-ng commented:
Still breaks on Travis and AppVeyor

jbenden commented on src/aircrack-util/crypto_engine_loader.c:
```diff
@@ -198,7 +198,7 @@ void (*dso_ac_crypto_engine_calc_mic)(ac_crypto_engine_t * engine,
 #endif
 
-#if defined(CYGWIN)
+#if defined(__CYGWIN__)
```
This macro is defined via the Autoconf scripts.

neheb replied:
Yeah it was an attempt to fix the buildbots. Apparently the reason they fail is that they pass -Werror, which is fairly amusing.

Speaking of which, the cygwin defines are all over the place.

jbenden replied:
Glad you like that. :)

jbenden reviewed (commented):
Uck!

jbenden reviewed (changes requested):
Please look at the quantity of files/lines changed with the diff. I'm sure it broke Windows line endings on certain files...

Could you revert to an older commit and try that change over?

neheb commented:
Yeah I'll fix it. I accidentally added unrelated stuff.

neheb committed:
Fix -Wmissing-declaration warnings

Improves C99 compatibility and should also help the compiler to produce
smaller code.

neheb commented:
Rebased. Single buildbot failure seems to be due to some license thing.
